Dispatcher.php

```diff
@@ -102,15 +114,32 @@ public function send(Message $message, $templateName = '', $templateContent = ar
 
         if (!empty($templateName)) {
             return $this->service->messages->sendTemplate($templateName, $templateContent, $message->toArray(), $async, $ipPool, $sendAt);
-
         }
 
         return $this->service->messages->send($message->toArray(), $async, $ipPool, $sendAt);
-
     }
 
+    private function useProxy()
+    {
+        return $this->proxy['use'];
+    }
 
-
-
-
+    private function addCurlProxyOptions()
+    {
+        if ($this->proxy['host'] !== null) {
+            curl_setopt($this->service->ch, CURLOPT_PROXY, $this->proxy['host']);
+        }
+        
+        if ($this->proxy['port'] !== null) {
+            curl_setopt($this->service->ch, CURLOPT_PROXYPORT, $this->proxy['port']);
+        }
+        
+        if ($this->proxy['user'] !== null && $this->proxy['password'] !== null) {
```

Comment:
No, because there is no way to add a user without password according to the official documentation http://php.net/manual/en/function.curl-setopt.php:
> ``CURLOPT_PROXYUSERPWD``: A username and password formatted as "[username]:[password]" to use for the connection to the proxy. 

You have to set both of them.